backend/controller/SequenceController.ts

```diff
@@ -0,0 +1,384 @@
+        }
+        const seqlist: ISequence[] = [];
+        for (const x of user.sharedSequencesReadAccess.concat(
+            user.sharedSequencesWriteAccess
+        )) {
+            const seq = await DBSequence.findOneBy({ code: x });
+            if (seq === null) {
+                // May just bring it back to consistency?
+                throw new LoernwerkError(
+                    'User has access to a non existing Sequence',
+                    LoernwerkErrorCodes.NOT_FOUND
+                );
+            }
+            seqlist.push(seq);
+        }
+        return seqlist;
+    }
+
+    /**
+     * Searches a sequence with corresponding code and reduces it to the attributes 'name, authorId, slideCount, code'. Throws an error if no sequence could be found.
+     * @param code the code of the sequence
+     * @returns the sequence (reduced)
+     */
+    public static async getSequenceForExecution(
+        code: string
+    ): Promise<Partial<ISequence>> {
+        const seq = await DBSequence.findOne({
+            where: { code: code },
+            select: {
+                name: true,
+                authorId: true,
+                slideCount: true,
+                code: true,
+            },
+        });
+        if (seq === null) {
+            throw new LoernwerkError(
+                'no sequence found',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        return seq;
+    }
+
+    /**
+     * Searches a slide of the sequence with given code. Throws an error if no sequence or slide could be found.
+     * @param code the code of the sequence
+     * @param slideIndex the index of the searched slide
+     * @returns the matching slide
+     */
+    public static async getSequenceSlideByCode(
+        code: string,
+        slideIndex: number
+    ): Promise<ISlide> {
+        const slide = await DBSlide.findOne({
+            where: { sequenceCode: code, order: slideIndex },
+        });
+        if (slide === null) {
+            throw new LoernwerkError(
+                'No matching Slide found',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        return slide;
+    }
+
+    /**
+     * stores the slides in the database
+     * @param slides the slides to store
+     */
+    private static async saveSlides(slides: ISlide[]): Promise<void> {
+        for (const x of slides) {
+            let slide = await DBSlide.findOneBy({ id: x.id });
+            if (slide === null) {
+                slide = new DBSlide();
+            }
+            slide.backgroundColor = x.backgroundColor;
+            slide.content = x.content;
+            slide.layout = x.layout;
+            slide.order = x.order;
+            slide.save();
+        }
+    }
+
+    /**
+     * Generates a random code for a sequence.
+     * @returns  the generated sequence
+     */
+    private static genCode(): string {
+        const codechar = 'abcdefghijklmnopqrstuvwxyz1234567890';
```

Comment:
Do we want lower or upper letters?
I would be in favor of upper

Reply:
doesnt matter, changed it to upper